src/kubernetes.cc

```diff
@@ -1052,6 +1278,14 @@ void KubernetesUpdater::StartUpdater() {
     pod_watch_thread_ = std::thread([=]() {
       reader_.WatchPods(watched_node, cb);
     });
+    if (config().KubernetesClusterLevelMetadata()) {
```

Comment:
I wonder if it makes sense to define an extra option just for service metadata? Or will we always want to retrieve it along with unscheduled pods?

Reply:
added a separate option. I wonder how the the two flags interact - i.e. if KubernetesClusterLevelMetadata is false, does it make sense for KubernetesServiceMetadata to be true

Reply:
Well, `KubernetesClusterLevelMetadata` means it's running the separate instance at the cluster level, rather than the per-node one. There are many things we can watch at cluster level. `KubernetesServiceMetadata` controls specifically whether we watch services/endpoints at cluster level. So `KubernetesServiceMetadata` is ignored when `KubernetesClusterLevelMetadata` is false, but it can certainly be set to true. See my other comment about guarding the watch threads.